src/api.js

```diff
@@ -10,6 +10,12 @@ export default class API extends mapboxgl.Control {
 
   constructor() {
     super();
+    this.types = {
```

Comment:
Constants will help us prevent bugs internally and externally. And I put them in API.js since its public-facing.